library/src/main/java/io/ona/kujaku/helpers/MapboxLocationComponentWrapper.java

```diff
@@ -0,0 +1,47 @@
+package io.ona.kujaku.helpers;
+
+import android.content.Context;
+import android.support.annotation.NonNull;
+
+import com.mapbox.android.core.location.LocationEngine;
+import com.mapbox.mapboxsdk.location.LocationComponent;
+import com.mapbox.mapboxsdk.location.modes.CameraMode;
+import com.mapbox.mapboxsdk.location.modes.RenderMode;
+import com.mapbox.mapboxsdk.maps.MapboxMap;
+
+import io.ona.kujaku.exceptions.LocationComponentInitializationException;
+
+/**
+ * @author Vincent Karuri
+ */
+public class MapboxLocationComponentWrapper {
+
+    private static LocationComponent locationComponent;
+
+    private static MapboxLocationComponentWrapper mapboxLocationComponentWrapper;
+
+    private MapboxLocationComponentWrapper() {}
+
+    @SuppressWarnings( {"MissingPermission"})
+    public static void init(@NonNull MapboxMap mapboxMap,@NonNull Context context) {
+        mapboxLocationComponentWrapper = new MapboxLocationComponentWrapper();
+        LocationComponent locationComponent = mapboxMap.getLocationComponent();
```

Comment:
Remove the definition of this private variable and assign the static variable `locationComponent`

Reply:
Any reason for this?

Reply:
the internal created a variable that does nothing, you just assign it to the global variable

Reply:
Removed.